### PR TITLE
Replaces simplexml with dom.

### PR DESCRIPTION
**Pull request by JojoVes**
**JIRA Ticket**: [ISLANDORA-2022](https://jira.duraspace.org/browse/ISLANDORA-2022)

# What does this Pull Request do?

This PR enables discrepancies in MODS info, specifically the dateIssued tag in Newspaper Issues, to be handled in a useful manner without getting confused by prefixes, or lack thereof.

# What's new?
It swaps out simplexml for dom.
When a MODS form doesn't have an originInfo tag, we now add one instead of crashing.

Example:
* If we do a Newspaper Batch ingest without MODS.xml files, or very sparse MODS.xml files, the module can add a date instead of crashing when entering a date in the your-newspaper/manage/newspaper.

# How should this be tested?

- A test batch with descriptions and expected behaviour is located in [ISLANDORA-2022](https://jira.duraspace.org/browse/ISLANDORA-2022)
- To reproduce the problem before testing this fix, create a Newspaper object, ingest an issue via Newspaper Batch without a MODS.xml file (you can extract issue2 from the test batch in the ticket), then try to enter a date in the manage/newspaper section of the paper containing that issue.

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

File: includes/utilities.inc
Comment (whikloj) [inline includes/utilities.inc]: This is not vital to the PR, but I think this loop of a loop was to work around a simplexml limitation. You might be able to drop the `foreach` on lines 326-328 just delete the nodes straight away using DOM.

File: includes/utilities.inc
Comment (qadan) [inline includes/utilities.inc]: Think getting this `$mods` copy is unnecessary; should be working with the DOM itself that's already been grabbed in `$dom`.

File: includes/utilities.inc
Comment (qadan) [inline includes/utilities.inc]: Don't think these getting-and-using-the-prefix businesses are really necessary; DOMDocument should take care of namespacing for you.

File: includes/utilities.inc
Comment (qadan) [inline includes/utilities.inc]: Ah, I guess we can remove this `if` check as well, no? Since `$dates` is properly `Traversable` and all

File: includes/utilities.inc
Comment (willtp87) [inline includes/utilities.inc]: Should always be TRUE, could be the return value from loadXML.  The change might regress the test case?

**Review by willtp87 — APPROVED** [review]
This is passing my checks.

**Comment by DiegoPino** [comment]
@whikloj you were reviewing this right? Looks fine on this side also, so if everything is addressed you are good to go. If not, I will go for it in 24hrs (hope you all are fine with this). This issue does not affect us (metro) but I just read (and answered to) someone in the list having difficulties with the no default namespace and prefixed namespace MODS thingy.

**Comment by whikloj** [comment]
@DiegoPino I have not had time to test all the use cases @JojoVes listed. So if you are good to go, then feel free.

**Comment by DiegoPino** [comment]
@whikloj oh, I neither tested all the uses cases,shame-on-me, just too many! I just tested what I know and read are the differences between the different DOM/XML objects and the way they handle XPaths, and how some of the issues got solved. But if you feel more comfortable, can(and will) do later today. Thanks a lot! Once I got that I will merge (if all works out)

**Comment by whikloj** [comment]
I'll spend some time on this today, as I am off on vacation (next week). Building up some test objects now.

**Comment by whikloj** [comment]
👍

**Pull request merged**
